homeassistant/helpers/config_validation.py

```diff
@@ -1648,7 +1648,7 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict:
         vol.Required(CONF_PLATFORM): str,
         vol.Optional(CONF_ID): str,
         vol.Optional(CONF_VARIABLES): SCRIPT_VARIABLES_SCHEMA,
-        vol.Optional(CONF_ENABLED): boolean,
+        vol.Optional(CONF_ENABLED): vol.Any(boolean, template_complex),
```

Comment:
I don't think it's correct to allow complex templates (i.e. a dictionary or list of templates) here

Reply:
I changed it to use simple `template`. However, this slightly changes the behaviour. With `template_complex` an explicit `enabled: None` value is valid, whereas with `template` it is not.
> Unnamed automation could not be validated and has been disabled: invalid boolean value None for dictionary value @ data['trigger'][2]['enabled']. Got None

Shall I add the explicit None value to the validator
```py
vol.Optional(CONF_ENABLED): vol.Any(boolean, template, None),
```
or keep it like this?

Reply:
The existing behavior is that a missing `enabled` flag is interpreted as `True`, and the flag if provided must be a `bool`.
This PR allows the flag to be either a `bool` or a template which renders to a bool. Why do we need to allow `None`?

Reply:
I don't think there is any reason to allow `None` explicitly. I'll change the example in the architecture discussion description accordingly (this uses `default: null` like the example above currently).